server/persistence/readclient/readclient_list.go

```diff
@@ -50,6 +51,17 @@ func (c *ReadClient) ListUserWorkspaces(
 	// filter by namespace
 	filterByNamespace(ww, listOpts.Namespace)
 
+	for i := range ww.Items {
+		// apply is-owner label
+		mutate.ApplyIsOwnerLabel(&ww.Items[i], user)
+
+		// apply has-direct-access label
+		err := mutate.ApplyHasDirectAccessLabel(ctx, c.internalClient, &ww.Items[i], user)
+		if err != nil {
+			return kerrors.NewInternalError(fmt.Errorf("error retrieving the list of workspaces for user %v", user))
+		}
+	}
```

Comment:
It seems to me this logic is reused multiple times. Would it make sense add an auxiliary function in the `mutate` pkg or a private `ReadClient`'s method?

Reply:
I'm thinking for now, leave it as-is, but we should consider an `ApplyMutator` if we need end up needing any more mutations.

Reply:
fixed in 08d6cb5